src/nrniv/nrnapi.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+#include "hocdec.h"
+#include "section.h"
```

Comment:
Is there an advantage to making this file be limited to C?

It's part of NEURON which is C++ not a file to import from C.

To me, the main advantage of declaring a C interface as here with the extern C is to allow basically everything to load it and eliminate name managing issues.

Reply:
I'm a little confused; I thought that when this work was discussed in past dev meetings the idea was that we wanted to define a C (as opposed to C++) API for NEURON, because there was interest in pursuing bindings to other software, and that those wanted to be implemented in C for `$reasons`.

Seeing the `nrnapi.h` in this PR, I assumed that this header was defining that C API and it would be installed alongside NEURON, and that any external project/bindings (potentially implemented in C) would include it -- much like the [`neuron_api.h`](https://github.com/mcdougallab/neuron-c-api-demos/blob/main/for-proposed-formal-api/neuron_api.h) that you linked to. For sure we do not want external projects/bindings to be shipping around their own declarations of NEURON methods, and I don't see why we would want this `nrnapi.h` and that `neuron_api.h` to coexist.

If by C bindings what we mean is a C++ API that happens to be declared `extern "C"` to avoid faffing around with mangled names, I guess that's also OK but it sounds like a slightly weird middle ground.

Reply:
So maybe the trick is to end up with a `neuron_api.h` or whatever that's installed on user's machines?

Reply:
Removed the internal header file and replaced it with one intended for users to build against.